pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -200,7 +200,7 @@ func remediate(r *ReconcileMachineHealthCheck, remediationStrategy healthcheckin
 		// apply remediation logic, if at least one condition last more than specified timeout
 		// specific remediation logic goes here
 		if unhealthyForTooLong(nodeCondition, conditionTimeout) {
-			switch remediationStrategy {
+			switch *remediationStrategy {
```

Comment:
This method now needs to check for nil. It should probably do that early on and set a variable that isn't this pointer to a default value in the case of nil, then switch on that.

Reply:
done